Mochi/src/HyperParameters.cpp

```diff
@@ -0,0 +1,53 @@
+#include "HyperParameters.hpp"
+
+
+/* Constants for algorithm names . */
+const string HyperParameters::ALGORITHM_NAME_ADAM = "ADAM";
```

Comment:
I'm no CPP programmer, just curious: why initialize the constants here instead of in the HPP file?

Reply:
I don't like it either but I looked it up and apparently this is the "correct" way of defining and declaring constants. If I try to initialized them in the class definition in HPP I get an error. e.g.: 

```bash
error: in-class initialization of static data member ‘const string HyperParameters::ALGORITHM_NAME_PA’ of non-literal type
```
I didn't want to use defines for this which is why I approached it with constant variables.